ConsulAndServiceDiscovery/README.md

```diff
@@ -0,0 +1,114 @@
+# Introduction
+
+This document describes the consul usage and service discovery guidelines. This
+document is sure to evolve as we gain more experience with running consul in
+production. If you have any questions please use the Wikia #consul chat channel.
+
+## Introduction to Consul and Service Discovery
+
+For an introduction to consul please read the [documentation
+provided](https://www.consul.io/intro/) by the creators of consul. Read that
+first.
+
+Services can be "discovered" via DNS or the HTTP API. As an example, consider a
+query for a discussion service node. When using the DNS interface one would
+query for `discussion.service.sjc.consul`. Below is an example:
+
+```
+dig @localhost -p 8600 discussion.service.sjc.consul
+```
+
+The answer section of the above query will provide the IP addresses of instances
+that have registered themselves as `discussion` "services" and are healthy. 
+
+If you want to also query for the port that the service is listening one
+add `SRV` to the end of the above `dig` query.
+
+Consul DNS addresses take the form `TAG.SERVICE.service.DATACENTER.consul`.
+
+A similar request for "healthy" instances of the discussion service could query
+the HTTP API using the following:
+
+```
+curl 'http://consul.service.sjc.consul:8500/v1/health/service/discussion?passing'
+```
+
+The `passing` query parameter will limit the output to "healthy" services.
+
+# Guidelines
+
+## DNS Schema
```

Comment:
I think this schema would fit much better when defining marathon ids.
For consul tags I think its too limiting  and it actually breaks some fundamental assumptions how consul services are registered i.e. even if it will work now it might break in the future.